great_expectations/metrics/metric.py

```diff
@@ -118,7 +117,7 @@ def id(self) -> MetricConfigurationID:
     def config(self) -> MetricConfiguration:
         return Metric._to_config(
             instance_class=self.__class__,
-            metric_value_set=frozenset(self.dict().items()),
+            metric_value_set=list(self.dict().items()),
```

Comment:
this change was required because the metric value from users is a set

Reply:
no - the issue is that we're asking users to pass in a `set` as a value input to the metric. if we really need to keep this caching behavior, we would need to mutate that to be a frozenset.

Reply:
I think the problem with removing the caching is that each time `Metric.config` is called the `MetricConfiguration` will have a new id. Can we save the `MetricConfiguration` to a private `_config` and return that if it is available?

Reply:
why would it have a new ID? are we expecting a metric to be mutated? i thought that just happens in the metrics internals, not in our API classes.

Reply:
Ok, I re-read the `MetricConfiguration` code and now see that the id should be the same. I tried and failed to test the caching. We should be able to easily test that the id doesn't change though. If you are going to remove the cache can you please add a test to `tests/metrics/test_metric.py` `TestMetricConfig` to assert that calling twice results in the same id?

Reply:
@NathanFarmer, I removed the `id` field from this class in my PR: https://github.com/great-expectations/great_expectations/pull/10971/files#diff-c1f77a9f3411e39ae24afca33308d6b78ea4b4d636f7db0b5424c395bfdaed95L114 since it wasn't being used anywhere. Is this problematic?

Reply:
No, I guess this isn't a problem for the consumers of this API. 🚢